Bottom: 6408eaf7b4b976f9567cb6f52f2e03e7d0978bae
Top:    8044ea0aadcd6087aeaa2e84d46a16960f504fb6
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-06-08 08:51:45 -0500

Refresh of buildbot-playbook

---
diff --git a/ansible-playbook/buildbot.yml b/ansible-playbook/buildbot.yml
new file mode 100644
index 00000000..e8629926
--- /dev/null
+++ b/ansible-playbook/buildbot.yml
@@ -0,0 +1,67 @@
+---
+
+- name: Install/Setup Docker images and containers on daedalus
+  hosts: daedalus
+  become: true
+  become_user: root
+  become_method: sudo
+  tags:
+  - buildbot
+  roles:
+  - name: Build and start the Buildbot master container
+    role: buildbot-master
+    tags:
+    - buildbot-master
+
+  - name: Add buildbot user to Gitlab
+    role: gitlab-user
+    vars:
+      new_user: buildbot
+      new_user_name: Build Robot
+      new_user_email: buildbot@novatech-llc.com
+      new_user_pass: "{{ lookup('password', '/dev/null') }}"
+
+- name: Setup Buildbot worker system
+  hosts: buildbot-worker
+  become: true
+  become_user: root
+  become_method: sudo
+  tags:
+  - buildbot
+  - buildbot-worker
+  vars:
+    buildbot_master: "{{ hostvars.buildbot_master.ip_addr }}"
+  roles:
+  - role: buildbot-worker-ntel
+  - role: buildbot-worker-ptxdist
+
+- name: Gitlab public keys for buildbot containers
+  hosts: daedalus
+  become: true
+  become_user: root
+  become_method: sudo
+  tags:
+  - buildbot
+  roles:
+  - name: Add buildbot_master pubkey to buildbot user
+    role: gitlab-sshkey
+    vars:
+      username: buildbot
+      pubkey: "{{ hostvars.buildbot_master.pubkey }}"
+      title: "buildbot_master"
+  - name: Add buildbot_worker_ntel pubkey to buildbot user
+    tags:
+    - buildbot
+    role: gitlab-sshkey
+    vars:
+      username: buildbot
+      pubkey: "{{ hostvars.buildbot_worker_ntel.pubkey }}"
+      title: "buildbot_worker_ntel"
+  - name: Add buildbot_worker_ptxdist pubkey to buildbot user
+    tags:
+    - buildbot
+    role: gitlab-sshkey
+    vars:
+      username: buildbot
+      pubkey: "{{ hostvars.buildbot_worker_ptxdist.pubkey }}"
+      title: "buildbot_worker_ptxdist"
diff --git a/ansible-playbook/site.yml b/ansible-playbook/site.yml
index fa22ed60..b7bee5ae 100644
--- a/ansible-playbook/site.yml
+++ b/ansible-playbook/site.yml
@@ -84,11 +84,9 @@
     role: docker-buildsystem
     tags:
     - buildsystem_container
-  - name: Build and start the Buildbot master container
-    role: buildbot-master
-    tags:
-      - buildbot
-      - buildbot-master
+
+- name: Setup the buildbot containers
+  import_playbook: buildbot.yml
 
 - name: Setup the Build System (Test Station) with repo data
   hosts: buildsystem
diff --git a/roles/buildbot-master/tasks/main.yml b/roles/buildbot-master/tasks/main.yml
index 88501072..ff25c6fb 100644
--- a/roles/buildbot-master/tasks/main.yml
+++ b/roles/buildbot-master/tasks/main.yml
@@ -31,4 +31,16 @@
     networks: '{{ buildbot_networks }}'
     volumes: '{{ buildbot_volumes }}'
     env: '{{ buildbot_env }}'
-    ports: '{{ container_port_map.buildbot.port_args }}'
+    ports: '{{ buildbot_port_args }}'
+
+- name: Grab ssh public key
+  command: >
+    docker exec '{{ buildbot_container_name }}' cat /home/buildbot/.ssh/id_rsa.pub
+  changed_when: false
+  register: output
+
+- name: Add buildbot_master host
+  add_host:
+    name: buildbot_master
+    pubkey: "{{ output.stdout }}"
+    ip_addr: "{{ buildbot_ip_addr }}"
